doc/application/application.rst

```diff
@@ -966,50 +966,60 @@ preferred, since it correctly handles dependencies between options.
+describe the hardware it runs on. This section describes how you can
+modify an application build's device tree using overlay files.
 
-The Zephyr build system takes a Device Tree configuration
-option's value from the first source in which it is specified, taken from the
-following available sources, in order:
+Overlay files, which customarily have the :file:`.overlay` extension,
+contain device tree fragments which add to or modify the device tree
+used while building a Zephyr application. To add an overlay file or
+files to the build, set the CMake variable :makevar:`DTC_OVERLAY_FILE`
+to a whitespace-separated list of your overlay files.
 
-#. An application's current DT configuration (i.e. the file
-   :file:`zephyr/<board>.dts.pre.tmp` in the build directory).
+The Zephyr build system begins creation of a device tree by running
+the C preprocessor on a file which includes the following:
 
-#. The application's DT configuration file(s) given by the
-   :makevar:`DTC_OVERLAY_FILE` variable, either as set explicitly by the user
-   or using one of the default values detailed below.
+#. Configuration options from :ref:`Kconfig <configuration>`.
 
-#. Common overlays conditionally included by :file:`dts/common/common.dts`.
+#. The board's device tree source file, which by default is the Zephyr
+   file :file:`boards/ARCHITECTURE/BOARD/BOARD.dts`. (This location
+   can be overridden by setting the :makevar:`DTS_SOURCE` CMake
+   variable.)
 
-#. The board's default DT configuration for the current :makevar:`BOARD`
-   setting (i.e. the :file:`boards/ARCHITECTURE/BOARD/BOARD.dts`
-   file in the Zephyr base directory).
+#. Any "common" overlays provided by the build system. Currently, this
+   is just the file :file:`dts/common/common.dts`. (The common
+   overlays can be overridden by setting the
+   :makevar:`DTS_COMMON_OVERLAYS` CMake variable.)
 
-#. The kernel's default DT configuration settings (i.e. the default value given
-   to the option in one of Zephyr's :file:`dtsi` files).
+   The file :file:`common.dts` conditionally includes device tree
+   fragments based on Kconfig settings. For example, it includes a
+   fragment for MCUboot chain-loading, located at
+   :file:`dts/common/mcuboot.overlay`, if
+   :option:`CONFIG_BOOTLOADER_MCUBOOT` is set.
 
-The Zephyr build system determines a value for :makevar:`DTC_OVERLAY_FILE` by
-checking the following until one is found, in order:
+#. Any file or files given by the :makevar:`DTC_OVERLAY_FILE` CMake
+   variable.
+
+The Zephyr build system determines :makevar:`DTC_OVERLAY_FILE` as
+follows:
 
 - Any value given to :makevar:`DTC_OVERLAY_FILE` in your application
   :file:`CMakeLists.txt` (**before including the boilerplate.cmake file**),
   passed to the the CMake command line, or present in the CMake variable cache,
-  takes precedence
+  takes precedence.
+
+- The environment variable :envvar:`DTC_OVERLAY_FILE` is checked
+  next. This mechanism is deprecated.
```

Comment:
If it's deprecated, should it even be listed here? Or maybe write this:

    - In previous releases, the environment variable :envvar:`DTC_OVERLAY_FILE`
      was checked next, but this mechanism is now deprecated and not supported.

Reply:
I'm listing it here because despite being deprecated, the mechanism is still supported / still works. If users have this environment variable set, it will affect their build. For that reason I think it's worth documenting the behavior to avoid confusion. OK with you?

Reply:
So you're "deprecated" comment is saying it may not work in the future then.  I get it.

Reply:
Right. I'll add a note saying that users shouldn't rely on this, and ought to migrate to the previous mechanism in the list.